tools/rest_multi_models.lua

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env lua
+--[[
+
+Author: Vincent Nguyen (Ubiqus)
+
+License MIT
+
+Usage
+  Change the section below server_cfg according to your settings
+  When querying, "id" : n , where n is the model index in your config.
+
+  th tools/rest_multi_models.lua -model -gpuid 1
+  query the server:
+  curl -v -H "Content-Type: application/json" -X POST -d '[{ "src" : "international migration" , "id" : 1 }]' http://127.0.0.1:7784/translator/translate
+
+
+]]
+
+require('onmt.init')
+
+local yaml = require 'yaml'
+
+local tokenizer = require('tools.utils.tokenizer')
+local BPE = require ('tools.utils.BPE')
+local restserver = require('tools.restserver.restserver')
+
+local cmd = onmt.utils.ExtendedCmdLine.new('rest_multi_models.lua')
+
+local server_options = {
+   {
+     '-port', 7784,
+     [[Port to run the server on.]],
+     {
+       valid = onmt.utils.ExtendedCmdLine.isUInt
+     }
+   },
+   {
+     '-withAttn', false,
+     [[If set returns by default attn vector.]]
+   },
+   {
+     '-unload_time', 10,
+     [[Unload ununsed model from memory after this time.]]
+   },
+   {
+     '-model_config', '',
+     [[Path to yaml configuration file.]],
+     {
+       valid = onmt.utils.ExtendedCmdLine.fileExists
+     }
+   }
+}
+
+cmd:setCmdLineOptions(server_options, 'Server')
+
+local opt_server = cmd:parse(arg)
+local fconfig = io.open(opt_server.model_config, "rb")
+assert(fconfig)
+local content_config = fconfig:read "*a"
+fconfig:close()
+
+local server_cfg = yaml.load(content_config)
+
+-- CAREFULL opt is a table of options here, ie a table of what opt is usually in onmt
+
+local opt = {}
+
+for i=1, #server_cfg do
+  local model_cmd = onmt.utils.ExtendedCmdLine.new('rest_translation_server.lua')
```

Comment:
actually, it is not relevant here since `model_cmd` is just simulating translation server cmdline